src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -3339,8 +3339,12 @@ internal static string GetDriveQualifiedPath(string path, PSDriveInfo drive)
                 }
                 else
                 {
-                    if (path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
-                        path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) 
+                    // Check if the path begins with "\" or "/" (UNC Path or Path in Unix).
+                    // Ignore if the path resolves to a drive path, this will happen when path is equal to "\\" or "/".
+                    // Drive path still need formatting, so treat them as relative.
+                    if ((path.StartsWith(StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) || 
+                         path.StartsWith(StringLiterals.AlternatePathSeparatorString, StringComparison.Ordinal)) &&
+                         path.Length > 1)
```

Comment:
Please revert the order of "path.Length > 1" and (path.StartsWith() || path.StartsWith()), so that we can avoid unnecessary string comparison under some circumstance.